src/kvstore/raftex/RaftPart.cpp

```diff
@@ -1162,7 +1167,14 @@ bool RaftPart::leaderElection() {
                 }
             }
             weight_ = 1;
-            sendHeartbeat();
+            sendHeartbeat().then([this] (folly::Try<AppendLogResult>&& t) {
```

Comment:
Could we unify the logic with leaseValid?

Reply:
Good suggestion

Reply:
before sendHeartBeat succeeded,  the lease should be invalid,  so we could unify it?

Reply:
Agreed, the lease should be invalid. I mean it seems we can not tell whether sendHeartBeat succeeded by `lastMsgAcceptedDur_`? 
So if a node restart very quickly and elected as leader, but it has not committed any logs in this term. So we need to reject read request before any log committed, because there could be some logs  of previous term which has not been committed. I suppose it is different from lease, in this case, `lastMsgAcceptedDur_` would be quite small, and `leaseValid` would return valid. WDYT?

Reply:
> I suppose it is different from lease, in this case, lastMsgAcceptedDur_ would be quite small, and leaseValid would return valid. 

I am confused,  why lastMsgAcceptedDur_ would be quite small,  if the heartbeat not been accepted by the quorum followers,  it should be larger than the lease interval.

Reply:
I mean if a node elected as leader, before the heartbeat was accepted by quorum, `lastMsgAcceptedDur_` could be less than lease interval

Reply:
lastMsgAcceptedDur_  equals "now - lastMsgAcceptedTime"，  why it could be less than lease interval

Reply:
When it start up or after reboot, `lastMsgAcceptedTime` could be quite close to now?

Reply:
lastMsgAcceptedTime  should be inited to be zero after reboot.

Reply:
You're right